JSONModel/JSONModel/JSONModel.m

```diff
@@ -692,6 +692,31 @@ -(void)__inspectProperties
             if (p && ![propertyIndex objectForKey:p.name]) {
                 [propertyIndex setValue:p forKey:p.name];
             }
+            
+            //generate custom setters
+            if (p) {
+                p.customSetters = [[NSMutableDictionary alloc] init];
+                for (Class allowedType in allowedJSONTypes) {
+                    NSString *className = NSStringFromClass([JSONValueTransformer classByResolvingClusterClasses:allowedType]);
+                    
+                    if (!p.customSetters[className]) {
```

Comment:
Because ```allowedJSONTypes``` contains cluster classes like ```NSNumber```  and ```NSDecimalNumber```. Do this check can reduce some duplicate selector generation and dictionary insert.